src/Integration.Vsix/Analysis/AnalysisConfigMonitor.cs

```diff
@@ -46,20 +48,30 @@ internal sealed class AnalysisConfigMonitor : IAnalysisConfigMonitor, IDisposabl
         [ImportingConstructor]
         public AnalysisConfigMonitor(IAnalysisRequester analysisRequester,
             IUserSettingsProvider userSettingsProvider, // reports changes to user settings.json
-            ILogger logger) : this(analysisRequester, userSettingsProvider, logger, ThreadHandling.Instance)
+            IActiveSolutionBoundTracker activeSolutionBoundTracker,
+            ILogger logger) : this(analysisRequester, userSettingsProvider, activeSolutionBoundTracker, logger, ThreadHandling.Instance)
         { }
 
         internal AnalysisConfigMonitor(IAnalysisRequester analysisRequester,
             IUserSettingsProvider userSettingsProvider,
+            IActiveSolutionBoundTracker activeSolutionBoundTracker,
             ILogger logger,
             IThreadHandling threadHandling)
         {
             this.analysisRequester = analysisRequester;
             this.userSettingsProvider = userSettingsProvider;
+            this.activeSolutionBoundTracker = activeSolutionBoundTracker;
             this.logger = logger;
             this.threadHandling = threadHandling;
 
             userSettingsProvider.SettingsChanged += OnUserSettingsChanged;
+            activeSolutionBoundTracker.SolutionBindingChanged += OnSolutionBindingChanged;
```

Comment:
Perhaps we should listen to the `SolutionBindingUpdated` event too. However, I'll create a separate ticket for that as it would be a separate bug.